### PR TITLE
feat(tools): add panel component

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by Sboonny — COMMENTED** [review]
Didn't object.assign it because the component isn't complicated and most of the time it didn't have all the other component, except for one.

If you want to make it possible to type `Panel.Something` I will do so, just don't think it's worth because it isn't used much.

Edit: changed my mind, I am making compound components, to see if I can actually do it.

**Comment by ahmadabdolsaheb** [comment]
Thanks for taking care of this component. Here are a few nitpicks:
1) There are few more applicable tests from [bootstrap](https://github.com/react-bootstrap/react-bootstrap/blob/bs3-dev/test/PanelSpec.js) that we could include in this component.
2) The bsStyle does not show up in the storybook controls when viewing the component story
3)  The heading needs padding

----
unrelated: It seems like the color: --background-info and --foreground-info etc are set up incorrectly and need to be swapped, but that is another issue all together.

**Comment by ahmadabdolsaheb** [comment]
> We don't have `collapse`, or different heading for title that aren't `h3`. Do I add check for rendering test?
Just checking if passing DOM properties work and that each component/sub-component returns the expected element.

File: tools/ui-components/src/panel/types.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+export interface PanelProps extends React.HTMLAttributes<HTMLDivElement> {
+  className?: string;
+  children?: React.ReactNode;
```
Comment (Sboonny) [inline tools/ui-components/src/panel/types.ts:5]:
```suggestion
```

**Review by ahmaxed — APPROVED** [review]

**Review by Sembauke — APPROVED** [review]

**Pull request merged**